lms/djangoapps/courseware/grades.py

```diff
@@ -25,10 +26,71 @@
 from .module_render import get_module_for_descriptor
 from submissions import api as sub_api  # installed from the edx-submissions repository
 from opaque_keys import InvalidKeyError
+from xblock.core import XBlock
 
 
 log = logging.getLogger("edx.courseware")
 
+GRADED_LOCATION_CATEGORIES = set(
+    cat for (cat, xblock_class) in XBlock.load_classes() if (
+        getattr(xblock_class, 'has_score', True) or getattr(xblock_class, 'has_children', True)
+    )
```

Comment:
I'm kind of curious, does something break if you don't use `getattr` and just went with:

```python
GRADED_LOCATION_CATEGORIES = {
    cat for (cat, xb_cls) in XBlock.load_classes() if xb_cls.has_score or xb_cls.has_children
}
```